source/coll_db.f90

```diff
@@ -0,0 +1,598 @@
+  call h5_writeData(setCollDB, 7, cdb_nColl, cdb_cBx(1:cdb_nColl))
+  call h5_writeData(setCollDB, 8, cdb_nColl, cdb_cBy(1:cdb_nColl))
+  call h5_finaliseWrite(setCollDB)
+
+  deallocate(fldCollDB)
+
+end subroutine cdb_writeDB_HDF5
+#endif
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-22
+!  Updated: 2019-03-22
+!  Add a family to the family database
+! ================================================================================================ !
+subroutine cdb_addFamily(famName, nSig, famID, fErr)
+
+  use crcoall
+
+  character(len=*),  intent(in)  :: famName
+  real(kind=fPrec),  intent(in)  :: nSig
+  integer,           intent(out) :: famID
+  logical,           intent(out) :: fErr
+
+  famID = cdb_getFamilyID(famName)
+  if(famID == -1) then
+    cdb_nfam = cdb_nFam + 1
+    call cdb_allocFam
+    cdb_famName(cdb_nFam) = famName
+    cdb_famNSig(cdb_nFam) = nSig
+    famID = cdb_nFam
+    fErr = .false.
+  else
+    write(lout,"(a)") "COLLDB> Warning Collimator family '"//trim(famName)//"' already exists."
+    fErr = .true.
+  end if
+
+end subroutine cdb_addFamily
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-21
+!  Updated: 2019-03-22
+!  Find a family in the database and returns its ID
+! ================================================================================================ !
+integer function cdb_getFamilyID(famName) result(famID)
+
+  character(len=*), intent(in) :: famName
+  integer i
+
+  famID = -1
+  if(cdb_nFam > 0) then
+    do i=1,cdb_nFam
+      if(cdb_famName(i) == famName) then
+        famID = i
+        exit
+      end if
+    end do
+  end if
+
+end function cdb_getFamilyID
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-21
+!  Updated: 2019-03-22
+!  Find a family in the database and returns its nsig
+! ================================================================================================ !
+real(kind=fPrec) function cdb_getFamilyNSig(famName) result(nSig)
+
+  use crcoall
+
+  character(len=*), intent(in) :: famName
+  integer famID
+
+  famID = cdb_getFamilyID(famName)
+  if(famID > 0) then
+    nSig = cdb_famNSig(famID)
+  else
+    write(lout,"(a)") "COLLDB> Warning No nsig value found for collimator family '"//trim(famName)//"'"
+    nSig = 0.0_fPrec
+  end if
+
+end function cdb_getFamilyNSig
+
+! ================================================================================================ !
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-03-21
+!  Updated: 2019-03-21
+!  Write family database to file
+! ================================================================================================ !
+subroutine cdb_writeFam
```

Comment:
The additional utilities routines in `coll_db` are being cleaned up in PR step 3. Removing this will just cause merge conflicts as I re-purpose some of this code.

Reply:
ok I will wait to go through step #3 then